src/updatev.jl

```diff
@@ -6,9 +6,12 @@
 Update the noise variances `M.v` with respect to data `Y` using `method`.
 Internally calls [`updatevl(_)`](@ref) to update each entry.
 """
-function updatev!(M::HePPCATModel,Y,method)
+function updatev!(M::HePPCATModel,Y,method, eps)
     for (l,Yl) in enumerate(Y)
         M.v[l] = updatevl(M.v[l],M.U,M.λ,Yl,method)
+        if M.v[l] < eps
+            M.v[l] = eps
+        end
```

Comment:
```suggestion
        M.v[l] = max(M.v[l], eps)
```

Reply:
Interestingly, these are not completely equivalent:
```julia
julia> let v=0.2, eps=NaN
           if v < eps
               v = eps
           end
           v
       end
0.2

julia> let v=0.2, eps=NaN
           v = max(v, eps)
           v
       end
NaN
```
The reason is that `max` uses `isless` and `isless` defines a total order with things like `NaN` "ordered after regular values" (https://docs.julialang.org/en/v1/base/math/#Base.max, https://docs.julialang.org/en/v1/base/base/#Base.isless).

The above scenario is admittedly pretty unlikely, but between the two I slightly prefer using `max` since the `NaN` (which should probably not be there) is more likely to be propagated and caught.